File: app/src/main/java/com/plcoding/oraclewms/api/FormField.java
package com.plcoding.oraclewms.api;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class FormField {

    @SerializedName("type")
    @Expose
    private String type;
    @SerializedName("value")
    @Expose
    private String value;
    @SerializedName("form_key")
    @Expose
    private String formKey;
    @SerializedName("form_value")
    @Expose
    private String formValue;
    @SerializedName("line_number")
    @Expose
    private Integer lineNumber;
    @SerializedName("cursor")
    @Expose
    private Boolean cursor;
    public String getType() {
        return type;
    }
    public void setType(String type) {
        this.type = type;
    }
    public String getValue() {
        return value;
    }
    public void setValue(String value) {
        this.value = value;
    }
    public String getFormKey() {
        return formKey;
    }
    public void setFormKey(String formKey) {
        this.formKey = formKey;
    }
    public String getFormValue() {
        return formValue;
    }
    public void setFormValue(String formValue) {
        this.formValue = formValue;
    }
    public Integer getLineNumber() {
        return lineNumber;
    }
    public void setLineNumber(Integer lineNumber) {
        this.lineNumber = lineNumber;
    }
    public Boolean getCursor() {
        return cursor;
    }
    public void setCursor(Boolean cursor) {
        this.cursor = cursor;
    }

}

File: app/src/main/java/com/plcoding/oraclewms/api/FormField.kt
package com.plcoding.oraclewms.api

import java.io.Serializable
import java.util.Objects

data class FormField(
    var type: String? = null,
    var value: String? = null,
    var form_key: String? = null,
    var form_value: String? = null,
    var line_number: Int? = null,
    var cursor: Boolean = false,
    var formatters: Formatters? = null,
    var field_formatters: FieldFormatters? = null,
    var flag: Boolean? = true,
    var option_number: Int = 1,
    var option_name: String? = null,
) : Serializable {
    override fun toString(): String {
        return form_key.toString()
    }

    override fun hashCode(): Int {
        return Objects.hashCode(form_key)
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val form = o as FormField
        return form_key.equals(form.form_key)
    }
}
